src/sentry/replays/endpoints/project_replay_video_details.py

```diff
@@ -60,9 +62,55 @@ def get(self, request: Request, project, replay_id, segment_id) -> HttpResponseB
             return self.respond({"detail": "Replay recording segment not found."}, status=404)
 
         video_io = BytesIO(video)
+
+        if range_header := request.headers.get("Range"):
+            ranges = parse_range_header(range_header)
+
+            iterator = iter_from_range(ranges, video_io)
+            status_code = 206
+
+            headers = {}
+            if len(ranges) == 1:
+                positions = ranges[0].make_range(len(video) - 1)
+                if positions is None:
+                    length = 0
+                    byte_range = "*"
+                else:
+                    length = positions[1] - positions[0] + 1
+                    byte_range = f"{positions[0]}-{positions[1]}"
+
+                headers["Content-Length"] = length
+                headers["Content-Range"] = f"bytes {byte_range}/{len(video)}"
+            else:
+                headers["Content-Length"] = len(video)
+        else:
+            iterator = iter(lambda: video_io.read(4096), b"")
+            status_code = 200
+            headers = {
+                "Content-Length": len(video),
+                "Content-Range": f"0-{len(video)-1}",
```

Comment:
should we be adding the content range header if we're returning a full response? the semantics seem to be only if we're retirng a 206 / partial, do we add range: https://www.rfc-editor.org/rfc/rfc9110#section-14.4